client/rest/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -125,21 +126,21 @@ public HttpEntity getEntity() {
     }
 
     /**
-     * Set the headers to attach to the request.
+     * Add the provided headers to the request.
      */
-    public void setHeaders(Header... headers) {
+    public void addHeaders(Header... headers) {
```

Comment:
I was undecided on leaving the varargs or not. This way it breaks existing code much less. And it is kind of convenient. But it can be seen as confusing I guess.